lib/vintage_net_wizard/activity_monitor.ex

```diff
@@ -0,0 +1,52 @@
+defmodule VintageNetWizard.ActivityMonitor do
+  @moduledoc false
+
+  # This server is used to watch for activity in the wizard. If the inactivity
+  # timeout is triggered this server will shutdown the wizard. In order to keep
+  # this server from shutting down the wizard call `ping/0` to reset the
+  # timeout. The timeout should be in minutes.
+
+  use GenServer
+
+  require Logger
+
+  @doc """
+  Start the watcher
+  """
+  @spec start_link(timeout()) :: GenServer.on_start()
+  def start_link(inactivity_timeout) do
+    GenServer.start_link(__MODULE__, inactivity_timeout, name: __MODULE__)
+  end
+
+  @doc """
+  Ping the watcher to prevent it shutting down the wizard
+  """
+  @spec ping() :: :ok
+  def ping() do
+    GenServer.call(__MODULE__, :ping)
+  end
+
+  @impl GenServer
+  def init(inactivity_timeout) do
+    {:ok, inactivity_timeout, timeout_to_milliseconds(inactivity_timeout)}
+  end
+
+  @impl GenServer
+  def handle_call(:ping, _from, inactivity_timeout) do
+    {:reply, :ok, inactivity_timeout, timeout_to_milliseconds(inactivity_timeout)}
+  end
+
+  @impl GenServer
+  def handle_info(:timeout, inactivity_timeout) do
+    Logger.info("[VintageNetWizard] inactivity timeout, shutting down wizard")
+    :ok = VintageNetWizard.stop_wizard()
+    {:stop, :normal, timeout_to_milliseconds(inactivity_timeout)}
```

Comment:
Not that it matters since it's stopping, but the third element is the state, inactivity_timeout.

Reply:
:+1: